Shopify/hydrogen
# Add command to list environments

### WHY are these changes introduced?

We're introducing the ability for developers to specify an environment to use variables from (https://github.com/Shopify/hydrogen/pull/883) but they'd need to open the admin to see which branches are assigned to which environments in order to do so.

### WHAT is this pull request doing?

Introduces a new `env list` command that will list all of the environments on the linked Hydrogen storefront that have a `branch`.

```
➜  demo-store git:(list-environments) ✗ h2 env list

Name        Branch   URL
──────────  ───────  ──────────────────────────────────────
Production  main     https://example.com
Staging     staging  https://staging-123.myshopify.dev
Queen       queen    https://queen-456.myshopify.dev
Preview     -        -
```

### HOW to test your changes?

1. Pull the branch and navigate to the `cli` directory
1. Run `npm run build`
1. Navigate to the demo store directory `cd ../../templates/demo-store`
1. Run `h2 env list` (note: if `h2` is unavailable run `npx shopify hydrogen shortcut` first) to see a list of your environments
   - You should be prompted to provide a shop
   - You should then be prompted to authenticate with that shop
   - You should be prompted that you need to link to a storefront
1. Confirm that the correct list of environments is shown

#### Checklist
